File: Joseffe Barroso/Java/Polimorfismo/src/Model/Conta.java
package Model;

import java.util.ArrayList;

public class Conta {
	private String agencia;
	private String numero;
    private double saldo;
    private ArrayList<String> historico;
    
    public ArrayList<String> getHistorico() {
		return historico;
	}

	public void setHistorico(ArrayList<String> historico) {
		this.historico = historico;
	}
	
	public String getAgencia() {
		return agencia;
	}
	public void setAgencia(String agencia) {
		this.agencia = agencia;
	}
	public String getNumero() {
		return numero;
	}
	public void setNumero(String numero) {
		this.numero = numero;
	}
	
	public Double getSaldo() {
        return saldo;
    }

    public void setSaldo(Double saldo) {
        this.saldo = saldo;
    }

    public void depositar(Double valor) {
    	historico.add("Depósito: +" + valor);
    	this.saldo = this.saldo + valor;
    }
    
    public Boolean sacar(Double valor) {
    	if (this.saldo > valor) {
        	this.saldo = this.saldo - valor;
        	historico.add("Saque: -" + valor);
        	return true;
    	} else {
    		return false;
    	}
    }
	
	public Conta () {
    	this.historico = new ArrayList<String>();	
	}
	
	public Conta (String agencia, String numero, Double saldo) {
		this.agencia = agencia;
		this.numero = numero;
		this.saldo = saldo;
	}
}

File: Joseffe Barroso/Python/Exercícios - POO/Herança/Banco/Conta.py
class Conta:
    def __init__(self, titular, numConta, saldo):
        self.__titular = titular
        self.__numConta = numConta
        self.__saldo = saldo
        self.__historico = []

    @property
    def titular(self):
        return self.__titular

    @titular.setter
    def titular(self, titular):
        self.__titular = titular

    @property
    def numConta(self):
        return self.__numConta

    @numConta.setter
    def numConta(self, numConta):
        self.__numConta = numConta

    @property
    def saldo(self):
        return self.__saldo

    @saldo.setter
    def saldo(self, valor):
        self.__saldo = valor

    def sacar(self, valor):
        if self.__saldo < valor:
            print("Saldo insuficiente.")
            return
        self.__saldo -= valor
        self.__historico.append(f"Saque de R$ {valor:.2f}")

    def depositar(self, valor):
        self.__saldo += valor
        self.__historico.append(f"Depósito de R$ {valor:.2f}")

    def extrato(self):
        print(f"Extrato da conta de {self.__titular}:")
        for transacao in self.__historico:
            print("-", transacao)
        print(f"Saldo atual: R$ {self.__saldo:.2f}")